book_source/03_topical_pages/03_pecan_xml.Rmd

```diff
@@ -768,14 +828,125 @@ The following tags can be used for state data assimilation. More detailed inform
+    <min_value>0</min_value>
+    <max_value>9999</max_value>
+   </variable>
+  </state.variables>
+</state.data.assimilation>
+```
+
+* **variable** : [required] Each of this section should include each state variables separately.
+* **variable.name** : [required] The name of the state variable (same as what we call it in the model output file. for example, "SIPNET.out" file).
+* **unit** : [required] The unit of the state variable (can be empty if it's unitless).
+* **min_value** : [required] The minimum range of the state variable.
+* **max_value** : [required] The maximum range of the state variable.
+
+### Observation Preparation for State Data Assimilation {#xml-observation-preparation}
+
+The following tags can be used for observation preparation for the SDA workflow.
+
+```xml
+<state.data.assimilation>
+  <Obs_Prep>
+   <Landtrendr_AGB>
+    <AGB_indir>/projectnb/dietzelab/dongchen/Multi-site/download_500_sites/AGB</AGB_indir>
+    <allow_download>TRUE</allow_download>
+    <export_csv>TRUE</export_csv>
+    <timestep>
+     <unit>year</unit>
+     <num>1</num>
+    </timestep>
+   </Landtrendr_AGB>
+   
+   <MODIS_LAI>
+    <search_window>30</search_window>
+    <export_csv>TRUE</export_csv>
+    <run_parallel>TRUE</run_parallel>
+    <timestep>
+     <unit>year</unit>
+     <num>1</num>
+    </timestep>
+   </MODIS_LAI>
+   
+   <SMAP_SMP>
+    <search_window>30</search_window>
+    <export_csv>TRUE</export_csv>
+    <update_csv>FALSE</update_csv>
+    <timestep>
+     <unit>year</unit>
+     <num>1</num>
+    </timestep>
+   </SMAP_SMP>
+   
+   <Soilgrids_SoilC>
+    <timestep>
+     <unit>year</unit>
+     <num>1</num>
+    </timestep>
+   </Soilgrids_SoilC>
+   <outdir>/projectnb/dietzelab/dongchen/All_NEON_SDA/test_OBS</outdir>
+   <start.date>2012-07-15</start.date>
+   <end.date>2021-07-15</end.date>
+  </Obs_Prep>
+</state.data.assimilation>
+```
+
+The `Obs_Prep` section is specifically designed for the SDA_OBS_Assembler function within the `state.data.assimilation` section, which helps construct the obs.mean and obs.cov objects more efficiently. The assembler function provides support for MODIS LAI, Landtrendr AGB, SMAP soil moisture profile, and Soilgrids soil C observations. Within the `Obs_Prep` section, the user needs to specify data streams that will be proceeded in the assembler function as `DataSource_AbbreviationOfData` (for example: `Landtrendr_AGB` for Landtrendr aboveground biomass observations, or `Soilgrids_SoilC` for soilgrids soil organic carbon concentration). Other than that, the output directory, and start and end date need to be set within the `Obs_Prep` section, note that if you choose to export csv file for any data, you will get the corresponding csv file just under the outdir, for the Rdata of obs.mean and obs.cov objects, they will be stored whthin the Rdata folder within the outdir, if the Rdata folder is not created, then the assembler will help to create this folder automatically. 
```

Comment:
In general this is good. I'd recommend one small tweak: break this into two paragraphs between the example and the "other than that" and add a bullet list of currently supported functions. If you do this you can also drop the sentence that lists the currently supported data sources. Explaining the naming convention and then listing the function names should be sufficient for now, while the converse (listing the sources, as is currently done) does create ambiguity about the exact function names.

One thing to keep in mind for the future -- instead of a bullet list it would be even better to include a table that lists not just the function name but also useful info like default spatial resolution, temporal resolution, units, etc. This info isn't required to use the functions but it is definitely something users will want to know, and something that will be helpful when setting the timestep.

Reply:
Fixed.